flax/traverse_util.py

```diff
@@ -77,7 +80,40 @@ class _EmptyNode:
 empty_node = _EmptyNode()
 
 
-def flatten_dict(xs, keep_empty_nodes=False, is_leaf=None, sep=None):
+@overload
+def flatten_dict(xs: Mapping[Any, Any],
+                 keep_empty_nodes: bool = False,
+                 is_leaf: None | Callable[[tuple[Any, ...],
+                                           Mapping[Any, Any]], bool] = None,
+                 sep: None = None
+                 ) -> dict[tuple[Any, ...], Any]:
+    ...
+
+@overload
+def flatten_dict(xs: Mapping[Any, Any],
+                 keep_empty_nodes: bool,
+                 is_leaf: None | Callable[[tuple[Any, ...],
+                                           Mapping[Any, Any]], bool],
+                 sep: str,
+                 ) -> dict[str, Any]:
+    ...
+
+@overload
+def flatten_dict(xs: Mapping[Any, Any],
+                 keep_empty_nodes: bool = False,
+                 is_leaf: None | Callable[[tuple[Any, ...],
+                                           Mapping[Any, Any]], bool] = None,
+                 *,
+                 sep: str
```

Comment:
why is this overload needed?

Reply:
I couldn't figure out how to support both:
`flatten_dict(xs, sep='/')` and `flatten_dict(xs, True, None, '/')`.  Do you know a way to do it one line?  Can I just get rid of the defaults when writing overloads?

Reply:
Another option would be to make keep_empty_nodes, is_leaf, and sep keyword-only parameters since it appears that they are always called this way.